Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkPipelineDetector.cs

```diff
@@ -38,22 +41,37 @@ private static void SetShaderKeywords()
                 if (material != null && IsTargetShader(material.shader))
                 {
 #if URP_AVAILABLE
-                    material.EnableKeyword("_UNIVERSAL_RENDER_PIPELINE");
-                    material.DisableKeyword("_HIGH_DEFINITION_RENDER_PIPELINE");
+                    bool shouldURP = true, shouldHDRP = false;
 #elif HDRP_AVAILABLE
-                    material.EnableKeyword("_HIGH_DEFINITION_RENDER_PIPELINE");
-                    material.DisableKeyword("_UNIVERSAL_RENDER_PIPELINE");
+                    bool shouldURP = false, shouldHDRP = true;
 #else
-                    material.DisableKeyword("_UNIVERSAL_RENDER_PIPELINE");
-                    material.DisableKeyword("_HIGH_DEFINITION_RENDER_PIPELINE");
+                    bool shouldURP = false, shouldHDRP = false;
 #endif
-                    EditorUtility.SetDirty(material);
-                    count++;
+                    // compare current with expected
+                    shouldURP = material.IsKeywordEnabled(URP) != shouldURP;
```

Comment:
Hmm.. won't this break when removing URP/HDRP in a project that previously had it?. The keyword will have already been enabled, but shouldURP will be false. So this line will boil down to `true != false`, which is true. Then it will set the keyword enabled below?

Reply:
Right. Not sure why I did that instead of it being inline in the IF. Fixed.